beta-src/src/utils/state/gameApiSlice/reducers/processMapClick.ts

```diff
@@ -0,0 +1,375 @@
+import { current } from "@reduxjs/toolkit";
+import TerritoryMap from "../../../../data/map/variants/classic/TerritoryMap";
+import BuildUnit from "../../../../enums/BuildUnit";
+import Territory from "../../../../enums/map/variants/classic/Territory";
+import { GameCommand } from "../../../../state/interfaces/GameCommands";
+import GameDataResponse from "../../../../state/interfaces/GameDataResponse";
+import GameOverviewResponse from "../../../../state/interfaces/GameOverviewResponse";
+import { GameState } from "../../../../state/interfaces/GameState";
+import OrdersMeta from "../../../../state/interfaces/SavedOrders";
+import { UnitSlotNames } from "../../../../types/map/UnitSlotName";
+import highlightMapTerritoriesBasedOnStatuses from "../../../map/highlightMapTerritoriesBasedOnStatuses";
+import getOrderStates from "../../getOrderStates";
+import processConvoy from "../../processConvoy";
+import resetOrder from "../../resetOrder";
+import setCommand from "../../setCommand";
+import startNewOrder from "../../startNewOrder";
+import updateOrdersMeta from "../../updateOrdersMeta";
+
+/* eslint-disable no-param-reassign */
+export default function processMapClick(state, clickData) {
+  const {
+    data: {
+      data: { currentOrders, contextVars },
+    },
+    order: {
+      inProgress,
+      method,
+      orderID,
+      onTerritory,
+      subsequentClicks,
+      toTerritory,
+      type,
+      unitID,
+    },
+    ordersMeta,
+    overview: {
+      user: { member },
+      phase,
+    },
+    territoriesMeta,
+  }: {
+    data: {
+      data: {
+        currentOrders: GameDataResponse["data"]["currentOrders"];
+        contextVars: GameDataResponse["data"]["contextVars"];
+      };
+    };
+    order: {
+      inProgress: GameState["order"]["inProgress"];
+      method: GameState["order"]["method"];
+      orderID: GameState["order"]["orderID"];
+      onTerritory: GameState["order"]["onTerritory"];
+      subsequentClicks: GameState["order"]["subsequentClicks"];
+      toTerritory: GameState["order"]["toTerritory"];
+      type: GameState["order"]["type"];
+      unitID: GameState["order"]["unitID"];
+    };
+    ordersMeta: OrdersMeta;
+    overview: {
+      user: { member: GameOverviewResponse["user"]["member"] };
+      phase: GameOverviewResponse["phase"];
+    };
```

Comment:
If you just typed data, order, and overview, then destructured their children after line 64, wouldnt the children be automatically typed without having to type all of them?